parser/lexer/lexer_test.go

```diff
@@ -1644,53 +2657,14 @@ func TestLexBlockComment(t *testing.T) {
 			[]token{
 				{
 					Token: Token{
-						Type: TokenBlockCommentStart,
-						Range: ast.Range{
-							StartPos: ast.Position{Line: 1, Column: 0, Offset: 0},
-							EndPos:   ast.Position{Line: 1, Column: 1, Offset: 1},
-						},
-					},
-					Source: "/*",
-				},
-				{
-					Token: Token{
-						Type: TokenBlockCommentContent,
-						Range: ast.Range{
-							StartPos: ast.Position{Line: 1, Column: 2, Offset: 2},
-							EndPos:   ast.Position{Line: 1, Column: 9, Offset: 9},
-						},
-					},
-					Source: `  // *X `,
-				},
-				{
-					Token: Token{
-						Type: TokenBlockCommentStart,
-						Range: ast.Range{
-							StartPos: ast.Position{Line: 1, Column: 10, Offset: 10},
-							EndPos:   ast.Position{Line: 1, Column: 11, Offset: 11},
-						},
-					},
-					Source: "/*",
-				},
-				{
-					Token: Token{
-						Type: TokenBlockCommentContent,
-						Range: ast.Range{
-							StartPos: ast.Position{Line: 1, Column: 12, Offset: 12},
-							EndPos:   ast.Position{Line: 1, Column: 17, Offset: 17},
-						},
-					},
-					Source: ` \\*  `,
-				},
-				{
-					Token: Token{
-						Type: TokenBlockCommentEnd,
+						Type:         TokenError,
+						SpaceOrError: MissingCommentEndError{},
 						Range: ast.Range{
-							StartPos: ast.Position{Line: 1, Column: 18, Offset: 18},
-							EndPos:   ast.Position{Line: 1, Column: 19, Offset: 19},
+							StartPos: ast.Position{Line: 1, Column: 20, Offset: 20},
+							EndPos:   ast.Position{Line: 1, Column: 20, Offset: 20},
 						},
 					},
-					Source: "*/",
+					Source: "\000",
```

Comment:
How come the source is the null character here?

Reply:
I guess I thought this is expected, since the error occurs at EOF (which doesn't have a position within the source code).

But I guess the idea here is to track the location of the last (unconsumed) token, which makes sense actually, 

Will update this.